src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -369,17 +369,17 @@ class ShenandoahHeap : public CollectedHeap {
 public:
   // Entry points to STW GC operations, these cause a related safepoint, that then
   // call the entry method below
-  void vmop_entry_init_mark();
-  void vmop_entry_final_mark();
+  void vmop_entry_init_mark(ShenandoahConcurrentMark* mark);
+  void vmop_entry_final_mark(ShenandoahConcurrentMark* mark);
```

Comment:
I find it odd that those entries now take ShenandoahConcurrentMark instance as argument, while all others don't. In particular because it implies it is the concurrent version anyway. Why is this necessary?

Reply:
ShenandoahConcurrentMark is stateless, can be created in place. The parameters are removed.